design/ironic-standalone-networking-service.md

```diff
@@ -0,0 +1,734 @@
+```
+
+For nodes with network configurations defined, the network interface driver
+should be set to `standalone-experimental`.
+
+### Ironic Networking Service Configuration
+
+Metal³ will generate a configuration file for the Ironic Networking service:
+
+```ini
+[DEFAULT]
+debug = <DEBUG_SETTING>
+rpc_transport = json-rpc
+auth_strategy = http_basic
+http_basic_auth_user_file = /etc/ironic/htpasswd
+
+[service_catalog]
+auth_type = http_basic
+username = admin
+password = <GENERATED_PASSWORD>
+endpoint_override = http://<IRONIC_API_INTERNAL_IP>:6385
+
+[networking_json_rpc]
+host_ip = <IRONIC_NETWORKING_SERVICE_IP>
+port = 8090
+auth_strategy = http_basic
+auth_type = http_basic
+http_basic_auth_user_file = /etc/ironic/htpasswd
+username = admin
+password = <GENERATED_PASSWORD>
+
+[networking]
+switch_config_file = /etc/ironic/networking/switch-configs.conf
+driver_config_dir = /etc/ironic/drivers
+enabled_switch_drivers = generic-switch
+serialize_switch_operations = True
+allowed_vlans =
+denied_vlans =
+allow_port_channels = False
+```
+
+### Switch Configuration File
+
+For each `BareMetalSwitch` resource, Metal³ will generate an entry in the
+switch configuration file:
+
+```ini
+[switch:<SWITCH_NAME>]
+driver_type = <DRIVER_TYPE>
+device_type = <DEVICE_TYPE>
+address = <ADDRESS>
+username = <USERNAME>
+password = <PASSWORD>
+enable_secret = <ENABLE_SECRET>
+mac_address = <MAC_ADDRESS>
+```
+
+This file will be stored in a Secret and mounted to the Ironic Networking
+container. When any `BareMetalSwitch` is created, updated, or deleted, the
+Secret will be regenerated and the Ironic Networking service will be restarted.
+
+### Ironic Port Configuration
+
+When a `BareMetalHost` has `networkInterfaces` defined, Metal³ will update the
+corresponding Ironic Port's `extra` field with a `switchport` dictionary
+conforming to the schema:
+
+```json
+{
+  "extra": {
+    "switchport": {
+      "mode": "access",
+      "native_vlan": 100,
+      "allowed_vlans": [200, 300, 400]
+    }
+  }
+}
+```
+
+This information represents one half of the details required to configure the
+switch port.  The identity of the switch and the attached port are obtained
+from LLDP and stored in the port's `local_link_connection` field, which is
+populated during initial inspection.
+
+### Implementation Details/Notes/Constraints
+
+1. **Service Network Setup**: Before creating `BareMetalHost` resources, users
+   must ensure that server switch ports are manually configured to the service
+   VLAN to allow initial communication with Ironic. This is a one-time setup
+   requirement. (This is known as the "idle" VLAN in Ironic terminology.)
+
+2. **Validation Timing**: Network interface validation can only occur after
+   inspection completes, as network interface information is not available
+   before that point. The `NetworkConfigurationReady` status field will be
+   updated after inspection.
```

Comment:
Let's be explicit that we depend on (in-band) inspection for all this work: some people disable inspection.

Reply:
Ok, will add.